torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -777,6 +777,79 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
   }
 
  public:
+  std::string moduleHierarchy() const {
+    std::string module_hierarchy("TOP");
+    std::vector<StackEntry> entries;
+    for (size_t i = 0; i < frames.size(); ++i) {
+      const Frame& frame = frames[i];
+      std::string fn_name = frame.function->function_name_;
+      // For each frame, type of the class with which the function is
+      // associated, is queried here. And the type name is added to
+      // module hierarchy.
+      const auto& g = frame.function->graph_;
+      std::string g_self_type;
+      if (g && g->inputs().size() > 0) {
+        const auto& g_self_type_ptr =
+            g->inputs()[0]->type()->cast<c10::ClassType>();
+        if (g_self_type_ptr) {
+          g_self_type = g_self_type_ptr->name()->qualifiedName();
+          g_self_type = g_self_type.substr(g_self_type.find_last_of('.') + 1);
+        }
+      }
+      module_hierarchy += "(" + g_self_type + ")::" + fn_name;
```

Comment:
not be a bad suggestion to use 'operator+=' or 'string::append()'

Reply:
Any particular reason as to why one vs. the other? Is it to avoid intermediate string construction?

Reply:
yep that's right, which is particularly relevant in the interpreter code.